app/controllers/api/v1/electricals_controller.rb

```diff
@@ -0,0 +1,63 @@
+module Api
+  module V1
+    class ElectricalsController < Api::V1::ApiController
+      load_and_authorize_resource :electrical, parent: false
+
+      resource_description do
+        short "Create, update and show an Electrical."
+        formats ["json"]
+        error 401, "Unauthorized"
+        error 404, "Not Found"
+        error 422, "Validation Error"
+        error 500, "Internal Server Error"
+      end
+
+      def_param_group :electrical do
+        param :electrical, Hash do
+          param :brand, String, desc: "name of the brand"
+          param :model, String, desc: "model of the record"
+          param :serial_number, String, desc: "serial number of the record"
+          param :standard, String, desc: "size of the record"
+          param :voltage, String, desc: "cpu of the record"
+          param :frequency, String, desc: "ram of the record"
+          param :power, String, desc: "hdd of the record"
+          param :system_or_region, String, desc: "optical of the record"
+          param :test_status, String, desc: "video of the record"
+          param :tested_on, Date, desc: "sound of the record"
+          param :updated_by_id, Integer, desc: "id of the user who updated the record"
+        end
+      end
+
+      def index
```

Comment:
were you going to create an `autocomplete` function? I can't remember what we decided

Reply:
No, I have kept it based on parameters. If we have `params[:distinct]` present, then only we will filter out the data. Otherwise will send all the electricals. 

There was a reason to not implement separate endpoint because, for some of the subforms(eg computer) there are lot of autocomplete fields and sending request everytime user types in the searchText would have reduced the overall performance. So to avoid that, we implemented it this way.